src/middleware/errorHandler.js

```diff
@@ -0,0 +1,20 @@
+import { AppError } from '../utils/errors.js';
+
+export const errorHandler = (err, req, res, next) => {
+  console.error(err);
+
+  if (err instanceof AppError) {
```

Comment:
You can add if statement for each of the error type here. and add a unique handler and you can modify the response as you need. For example for validation error you can add validation errors to the response.

Actually this what you did is also okay, as it will include the correct status code. But maybe at least for validation you can add a separate if statement